R/demean.R

```diff
@@ -296,19 +319,24 @@ degroup <- function(x,
     select <- c(interactions_no, colnames(new_dat))
   }
 
-  not_found <- setdiff(select, colnames(x))
+  not_found <- setdiff(c(select, by), colnames(x))
 
   if (length(not_found) && isTRUE(verbose)) {
     insight::format_alert(
-      sprintf(
-        "%i variables were not found in the dataset: %s\n",
-        length(not_found),
-        toString(not_found)
-      )
+      paste0(
```

Comment:
@strengejacke regarding your comment, I think it should be an error, yes. In most of other places in `datawizard` it throws an error, with rare exceptions where it is a warning (sometimes we give an argument `safe` to determine if it's an error or a warning)